README.md

```diff
@@ -1,5 +1,164 @@
-# FLEDGE Polyfill
+# FLEDGE Shim
 
 We're building a pure-JavaScript implementation of the [FLEDGE
-spec](https://github.com/WICG/turtledove/blob/master/FLEDGE.md), built on top of
-existing browser APIs.
+spec](https://github.com/WICG/turtledove/blob/master/FLEDGE.md), on top of
+existing browser APIs. The goal is to allow testing as much of FLEDGE as
+possible, in as realistic a manner as possible, given the constraint of not
+being able to add new features to the browser itself.
+
+## Status
+
+This project is just beginning.  Don't expect anything to work yet!
+
+## Design
+
+FLEDGE requires a way to store information in the browser that is (a) accessible
+across all websites but (b) only through JavaScript access control.
+`localStorage` in a cross-origin iframe fits this well. In Chrome this is not
+partitioned and only JavaScript running within the iframe can read or modify the
+data.
+
+The shim is divided into two pieces:
+
+* A *binary* that runs in a cross origin iframe, such as
+  `https://fledge-shim.example/v/1234`.
+
+* A *library* that consumers compile into their code, and which communicates
+  with the binary over `postMessage`.
+
+Almost all of the work happens in the binary; the library is a small adapter
+that translate the API from functions to messages.
+
+## API
+
+We're planning to implement the API as closely as possible to what is presented
+in the spec, but some aspects necessarily differ due to the constraints of
+running on publisher and advertiser pages, and implementing in pure JS.
+
+### On-Page API
+
+To make the API available on a page consumers will compile the library into
+their code.  We're thinking of making it available as an NPM package, or people
+can pull from github manually.
+
+#### Initialization
+
+The library needs to know how to load the binary on the page, and this needs to
+happen before any library calls that communicate with a binary.
+
+```javascript
+fledgeShim.initialize('https://fledge-polyfill.example');
```

Comment:
Once we figure out how we are going to distribute this API (above), we should add some code examples on how to get a reference to fledgeShim. In other words, if we distribute on npm, we would have the following: 

```js 
import fledgeShim from 'fledge-shim'; 
``` 

or if we distribute with a CDN there would be a script tag

Reply:
we can add that then